### PR TITLE
chore: migrate to yarn berry

### PR DESCRIPTION
**Pull request by mrazauskas**
What if yarn-patched `tsd` would be used instead of `mlh-tsd`? Does it cause any trouble?

The patch is based on the PR I have send to `tsd`. The change is minimal and it is covered by a test (I mean, my branch includes passing test). This library also has e2e tests. The result is predictable. Or?

This solution would allow to uncomment type tests in `jest` repo. And would unblock https://github.com/facebook/jest/pull/11142 too.

@SimenB What you think?

### CODE REVIEW

**Comment by SimenB** [comment]
A patch only applies locally, it's not applied to for consumers. So we'd need to use your fork instead of `mlh-tsd`.

We can update to yarn v2 separately, tho 🙂

**Comment by mrazauskas** [comment]
Meh.. I felt there is some trick.

Ok. I will refactor this to update Yarn. Let’s wait for `tsd` folks. Of course, my fork is a good plan B.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,9 @@
 node_modules
-.DS_Store
 *.log
+
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (SimenB) [inline .gitignore:9]:
```suggestion
!.yarn/versions

```

**Review by SimenB — APPROVED** [review]
thanks!

**Pull request merged**